### PR TITLE
Some fix about tow factor authentication

### PR DESCRIPTION
**Pull request by sbrunner**
* Fix some unneeded fail
* After first login (is_password_changed == False) we are not consider as logged in (no Cookie)
* On loginchange we should have to received the username in login field.

### CODE REVIEW

File: doc/developer/webservices.rst [hunk 1]
```diff
@@ -288,6 +288,16 @@ Connected JSON result
    }
 
 
+Init tow factor authentication JSON result
```
Comment (fredj) [inline doc/developer/webservices.rst]:
```suggestion
Init two factor authentication JSON result
```

**Comment by sbrunner** [comment]
@fredj Do you see that I add login int the post of the loginchange form is it OK to modify the UI with that? Or should we pass it in the query string?

File: doc/developer/webservices.rst [hunk 2]
```diff
@@ -273,10 +274,8 @@ Connected JSON result
 
    {
        "username": "<username>",
-       "is_intranet": true,
-       "is_password_changed": true/false, # If false the user should change his password
+       "is_intranet": true/true,
        "two_factor_enable": true/false, # Is the two factors authentication enabled?
```
Comment (jwkaltz) [inline doc/developer/webservices.rst]: two-factor authentication

File: doc/developer/webservices.rst [hunk 3]
```diff
@@ -287,6 +286,53 @@ Connected JSON result
...
+
+Login to the application.
+
+URL: ``.../login``
+
+Parameters (post form):
+
+* ``login``
+* ``password``
+* ``otp``: The secound factor code
```
Comment (jwkaltz) [inline doc/developer/webservices.rst]: second

File: doc/developer/webservices.rst [hunk 4]
```diff
@@ -287,6 +286,53 @@ Connected JSON result
...
+* 302 Found: Success with providing ``came_from`` parameter.
+* 400 Bad request: When ``login`` or ``password`` is missing.
+* 401 Unauthorized: On login failed.
+
+
+Login successful
+~~~~~~~~~~~~~~~~
+
+
+Init without two factor authentication JSON result
```
Comment (jwkaltz) [inline doc/developer/webservices.rst]: two-factor authentication

File: doc/developer/webservices.rst [hunk 5]
```diff
@@ -287,6 +286,53 @@ Connected JSON result
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code::
+
+   {
+       "is_password_changed": false, # Allways false
```
Comment (jwkaltz) [inline doc/developer/webservices.rst]: always

File: doc/developer/webservices.rst [hunk 6]
```diff
@@ -287,6 +286,53 @@ Connected JSON result
...
+       "two_factor_enable": false # Allways false
```
Comment (jwkaltz) [inline doc/developer/webservices.rst]: always

File: doc/developer/webservices.rst [hunk 7]
```diff
@@ -287,6 +286,53 @@ Connected JSON result
...
+   }
+
+
+Init two factor authentication JSON result
```
Comment (jwkaltz) [inline doc/developer/webservices.rst]: two-factor authentication

File: doc/developer/webservices.rst [hunk 8]
```diff
@@ -287,6 +286,53 @@ Connected JSON result
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code::
+
+   {
+       "two_factor_totp_secret": "<secret>", # The two factors authentication secret on first login
```
Comment (jwkaltz) [inline doc/developer/webservices.rst]: two-factor authentication

File: doc/developer/webservices.rst [hunk 9]
```diff
@@ -287,6 +286,53 @@ Connected JSON result
...
+       "otp_uri": "The OTM URI"
+       "is_password_changed": false, # Allways false
```
Comment (jwkaltz) [inline doc/developer/webservices.rst]: Always

File: doc/developer/webservices.rst [hunk 10]
```diff
@@ -287,6 +286,53 @@ Connected JSON result
...
+       "two_factor_enable": true # Allways true
```
Comment (jwkaltz) [inline doc/developer/webservices.rst]: Always

File: geoportal/c2cgeoportal_geoportal/views/entry.py [hunk 11]
```diff
@@ -1259,29 +1280,42 @@ def loginchange(self):
...
+        try:
+            user_obj = self.request.get_user(user)
+        except NoResultFound:  # pragma: no cover
+            LOG.info("The login '%s' does not exist.", user)
+            raise HTTPBadRequest("See server logs for details")
+
+        if self.two_factor_auth:
+            if not self._validate_2fa_totp(user_obj, self.request.POST.get("otp")):
+                if user is None:
+                    LOG.info("The secound factor is wrong for user '%s'.", user)
```
Comment (jwkaltz) [inline geoportal/c2cgeoportal_geoportal/views/entry.py]: second

File: geoportal/c2cgeoportal_geoportal/views/entry.py [hunk 12]
```diff
@@ -1249,6 +1269,7 @@ def loginuser(self):
     def loginchange(self):
```
Comment (arnaud-morvan) [inline geoportal/c2cgeoportal_geoportal/views/entry.py]: Not in PR scope, but loginchange => change_password, if we do not rename this method, a little docstring could be welcome.

Comment (sbrunner) [reply geoportal/c2cgeoportal_geoportal/views/entry.py]: +1, done :-)

**Pull request merged**